projects/common/src/telemetry/user-telemetry.module.ts

```diff
@@ -0,0 +1,45 @@
+import { ErrorHandler, Inject, ModuleWithProviders, NgModule } from '@angular/core';
+import { TelemetryGlobalErrorHandler } from './error-handler/telemetry-global-error-handler';
+import { UserTelemetryRegistrationConfig, USER_TELEMETRY_PROVIDER_TOKENS } from './telemetry';
+import { TrackDirective } from './track/track.directive';
+import { UserTelemetryInternalService } from './user-telemetry-internal.service';
+
+@NgModule({
+  declarations: [TrackDirective],
+  exports: [TrackDirective],
+  providers: [
+    {
+      provide: USER_TELEMETRY_PROVIDER_TOKENS,
+      useValue: [],
+      multi: true
+    },
+    {
+      provide: ErrorHandler,
+      useClass: TelemetryGlobalErrorHandler
+    }
+  ]
+})
+// tslint:disable-next-line: no-unnecessary-class
+export class UserTelemetryModule {
+  public constructor(
+    userTelemetryInternalService: UserTelemetryInternalService,
+    @Inject(USER_TELEMETRY_PROVIDER_TOKENS) providerConfigs: UserTelemetryRegistrationConfig<unknown>[][]
+  ) {
+    userTelemetryInternalService.register(...providerConfigs.flat());
```

Comment:
The way this is currently set up, we'll have dupe registrations since `USER_TELEMETRY_PROVIDER_TOKENS` is multi and will keep growing with each call to `withProviders`. Either need register to handle that or a way to dedupe like tracking last index.

Reply:
Lemme check